dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/RelationshipTypeObjectBundleHook.java

```diff
@@ -224,18 +195,28 @@ private void validateTrackedEntityInstance( RelationshipConstraint constraint, C
 
         if ( trackerDataView != null )
         {
-            trackerDataViewAttributes = trackerDataView.getTrackedEntityAttributes()
-                .stream().filter( Objects::nonNull ).map( BaseIdentifiableObject::getUid )
-                .collect( Collectors.toList() );
+            trackerDataViewAttributes = trackerDataView.getAttributes();
         }
 
         TrackedEntityType trackedEntityType = constraint.getTrackedEntityType();
 
+        if ( constraint.getProgramStage() != null )
+        {
+            addReports.accept( new ErrorReport( RelationshipConstraint.class, ErrorCode.E4023, PROGRAM_STAGE,
+                RELATIONSHIP_ENTITY, TRACKED_ENTITY_INSTANCE ) );
+        }
+
+        if ( constraint.getProgram() != null )
```

Comment:
We need to allow program to be specified for tracked entity types

Reply:
resolved